app/controllers/app.go

```diff
@@ -1,11 +1,139 @@
 package controllers
 
-import "github.com/revel/revel"
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"github.com/revel/revel"
+)
+
+type Defaults struct {
+	github_client_id     string
+	github_client_secret string
+	github_redirect_url  string
+	github_oAuth_url     string
+	github_api_user      string
+	github_login_url     string
+}
+
+func GetDefaults() Defaults {
+	defaults := Defaults{}
+	defaults.github_client_id = "GITHUB CLIENT ID"      //SET YOUR GITHUB CLIENT ID
+	defaults.github_client_secret = "GITHUB SECRET KEY" //SET SECRET KEY FOR GITHUB ACCOUNT
+	defaults.github_redirect_url = "http://localhost:9000/callback"
+	defaults.github_oAuth_url = "https://github.com/login/oauth/access_token"
+	defaults.github_api_user = "https://api.github.com/user"
+	defaults.github_login_url = "https://github.com/login/oauth/authorize?scope=user:email&client_id="
+	return defaults
+}
 
 type App struct {
 	*revel.Controller
 }
 
 func (c App) Index() revel.Result {
+	loginUrl := GetDefaults().github_login_url + GetDefaults().github_client_id
+	return c.Render(loginUrl)
+
+}
+
+func (c App) GitLogin() revel.Result {
+	c.Params.Query = c.Request.URL.Query()
+	fmt.Println(c.Params.Query)
```

Comment:
yup , learning a way to redirect these error message and render them into the view , but still could not find any good example, today or tomm, I will go through some sample projects in revel i have downloaded to learn more about this.